File: 727/lib/remove_duplicates.rb
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
  return 0 if nums.empty?
  k = 0
  nums[1..].each do |v|
    if nums[k] != v
      k += 1
      nums[k] = v
    end
  end
  k + 1
end

File: 727/src/removeDuplicates.js
/**
 * @param {number[]} nums
 * @return {number}
 */
removeDuplicates = nums => {
  if (!nums.length) return 0;
  let k = 0;
  nums.slice(1).forEach(v => nums[k] !== v && (nums[++k] = v));
  return k + 1;
}

module.exports = removeDuplicates;
